frontend/src/components/authentication/AuthenticationBarrier.tsx

```diff
@@ -0,0 +1,83 @@
+import { AuthenticationContext } from "@/contexts/AuthenticationContext";
+import { UserRole } from "@/i18n/user-role-messages";
+import { useRouter } from "next/router";
+import { useContext, useEffect, useMemo } from "react";
+import DisableSSR from "../next/DisableSSR";
+
+const allowedRoutesForUnauthenticatedUsers = [
+  "/_error",
+  "/login",
+  "/login/oidc-redirect",
+  "/login/admin",
+  "/login/teacher",
+  "/login/student",
+  // this page has a special role in the process of student authentication
+  "/session/[sessionId]/join",
```

Comment:
```suggestion
  "/login/student",
  
  // this page has a special role in the process of student authentication
  "/session/[sessionId]/join",
```

you might also want to clarify what you mean by "special", as it comes across as very obscure. it would be better to say _how_ it is special, e.g. "this page is used to sign-up students and create a class-specific identity" if that's the correct description :-)